src/hyperfine/benchmark.rs

```diff
@@ -234,13 +234,14 @@ pub fn run_benchmark(
     );
 
     // Run init command
-    let prepare_cmd = options
-        .preparation_command
-        .as_ref()
-        .map(|preparation_command| match cmd.get_parameter() {
-            Some((param, value)) => Command::new_parametrized(preparation_command, param, value),
-            None => Command::new(preparation_command),
-        });
+    let mut prepare_cmd = None;
+    if let Some(values) = &options.preparation_command {
+        if values.len() == 1 {
+            prepare_cmd = Some(new_command_from_get_parameter(cmd, &values[0]));
+        } else {
+            prepare_cmd = Some(new_command_from_get_parameter(cmd, &values[num]));
+        }
```

Comment:
how about first storing `&values[0]` or `&values[num]` in a local variable. Then, you only need to call `new_command_from…` once. If this works, we could remove that function again and inline it here.